narwhals/_spark_like/utils.py

```diff
@@ -110,9 +111,18 @@ def narwhals_to_native_dtype(
 
 
 def parse_exprs_and_named_exprs(
-    df: SparkLikeLazyFrame, *exprs: SparkLikeExpr, **named_exprs: SparkLikeExpr
+    df: SparkLikeLazyFrame,
+    *exprs: SparkLikeExpr,
+    with_columns_context: bool,
```

Comment:
I guess one bug could be if someone names a column `with_columns_context` and actually same for `df`

Reply:
we could address this by making `df` and `with_columns_context` positional-only arguments?

Reply:
Just read this comment, I still think it would not solve the issue.
I made a fix in #1871